### PR TITLE
Donation tracker

### PR DESCRIPTION
**Pull request by QuinnNgo**

### CODE REVIEW

File: components/DonationTracker/FamilyCard.jsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+const FamilyCard = ({ classes, family }) => {
+  return (
+    <div className={classes.container}>
+      <div className={classes.left}>
+        <div className={classes.yourDonationSupports}>
+          YOUR DONATION SUPPORTS
+        </div>
+        <div className={classes.familyName}>The {family.lastName} Family</div>
+        <Link href={`/give/${family.familyId}`}>
```
Comment (Raghav-Maheshwari) [inline components/DonationTracker/FamilyCard.jsx]: I believe we store the familyId as `beneficiaryId`

File: components/DonationTracker/DonationItem.jsx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    marginRight: '24px',
+  },
+  tracker: {
+    display: 'flex',
+    marginTop: '24px',
+    '& > $trackerItem': {
+      backgroundColor: '#BABABA',
+    },
+    '& > $trackerItem$completed': {
+      backgroundColor: '#10A0CF',
```
Comment (Raghav-Maheshwari) [inline components/DonationTracker/DonationItem.jsx]: let's use constants for the background colors

File: components/DonationTracker/DonationItem.jsx [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+        </div>
+        <div
+          className={classNames(classes.tracker, {
+            [classes.finished]: numCompletedStages === 4,
+          })}
+        >
+          {[1, 2, 3].map(stageNumber => (
+            <div
+              className={classNames(classes.trackerItem, {
+                [classes.completed]: stageNumber <= numCompletedStages,
```
Comment (Raghav-Maheshwari) [inline components/DonationTracker/DonationItem.jsx]: the way you implemented this is clean 🔥

File: components/DonationTracker/DonationItem.jsx (hunk 2 above)
Comment (QuinnNgo) [reply components/DonationTracker/DonationItem.jsx]: Can we standardize the names of these with design? It looks like Spence is using some preset color palette
image

I think we should do a full sweep of the entire codebase in a separate PR. For now, I've gone ahead and replaced all of the colors that have existing constants.

File: components/DonationTracker/DonationItem.jsx [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+  return (
+    <div className={classes.container}>
+      <div className={classes.metadataContainer}>
+        <img className={classes.image} src={item.image} alt={item.name} />
+        <div className={classes.statusContainer}>
+          <div>
+            <div className={classes.itemName}>{item.name}</div>
+            <div className={classes.statusDescription}>{statusDescription}</div>
+            <div className={classes.date}>
+              {moment(item.requestedTimestamp).format('L LT')}
```
Comment (Raghav-Maheshwari) [inline components/DonationTracker/DonationItem.jsx]: So we don't actually want to be using `item.requestedTimestamp` anywhere on the donation tracker - we want to show `donationTimestamp` for when we process their donation, `ready_for_pickup_timestamp` when the item is ready for pickup, `picked_up_timestamp` when the item has been picked up. Maybe we can use `item.requestedTimestamp` as a stopgap if the other values are null (for a item that doesn't have those values, event though every donation henceforth will have values for those timestamps) @QuinnNgo

**Pull request merged**